core/src/main/java/org/apache/druid/data/input/IntermediateRowParsingReader.java

```diff
@@ -93,39 +116,79 @@ public InputRow next()
       @Override
       public void close() throws IOException
       {
-        intermediateRowIterator.close();
+        intermediateRowIteratorWithMetadata.close();
       }
     };
   }
 
   @Override
   public CloseableIterator<InputRowListPlusRawValues> sample() throws IOException
   {
-    return intermediateRowIterator().map(row -> {
```

Comment:
could we just use regular `hasNext, next()` etc and get rid of `mapWithMetadata` entirely?

Reply:
If we use that, then the errors generated in the sample() call won't have the additional metadata.

Reply:
Removed the one-off implementation, and pushed it to the `IntermediateRowParsingReader` itself.